modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java

```diff
@@ -2443,6 +2505,8 @@ private void checkIncrementalCanBeCreated(
     ) throws IgniteCheckedException, IOException {
         File snpDir = snapshotLocalDir(name, snpPath);
 
+        ensureHardLinkAvailable(cctx.wal().archiveDir().toPath(), snpDir.toPath());
```

Comment:
Return of `archiveDir`is marked with `@Nullable`, also this thread doesn't check WAL for existence prior this line. So let's also add checks in this method: for WAL is available, and for archive dir isn't null.